pos_sale_order/point_of_sale.py

```diff
@@ -0,0 +1,239 @@
+            and order.partner_id.property_account_receivable.id)
+            or (account_def and account_def.id) or False)
+
+        if not args['account_id']:
+            if not args['partner_id']:
+                msg = _('There is no receivable account defined '
+                        'to make payment.')
+            else:
+                msg = _('There is no receivable account defined '
+                        'to make payment for the partner: "%s" (id:%d).') % (
+                            order.partner_id.name, order.partner_id.id,)
+            raise Warning(_('Configuration Error!'), msg)
+
+        context.pop('pos_session_id', False)
+
+        journal_id = data.get('journal', False)
+        statement_id = data.get('statement_id', False)
+        assert journal_id or statement_id, 'No statement_id '
+        'or journal_id passed to the method!'
+
+        for statement in order.session_id.statement_ids:
+            if statement.id == statement_id:
+                journal_id = statement.journal_id.id
+                break
+            elif statement.journal_id.id == journal_id:
+                statement_id = statement.id
+                break
+
+        if not statement_id:
+            raise Warning(_('Error!'),
+                          _('You have to open at least one cashbox.'))
+
+        args.update({
+            'statement_id': statement_id,
+            'journal_id': journal_id,
+            'ref': order.session_id.name,
+            'sale_ids': [(6, 0, [order_id])]
+        })
+
+        return args
+
+    @api.multi
+    def add_payment(self, order_id, data):
+        """Create a new payment for the order"""
+        statement_line_obj = self.env['account.bank.statement.line']
+        args = self._prepare_payment_vals(order_id, data)
+        statement_line_obj.create(args)
+
+        return args['statement_id']
+
+
+class PosSession(models.Model):
+    _inherit = 'pos.session'
+
+    def _get_domains(self, vals, partner_id, session):
+        vals = {[
+            ('session_id', '=', session.id),
+            ('state', '=', 'manual'),
+            ('partner_id', '=', partner_id),
+        ]}
+        return vals
+
+    @api.multi
+    def _confirm_orders(self):
+        sale_obj = self.env['sale.order']
+        for session in self:
+            order_ids = []
+            partner_id = session.config_id.anonymous_partner_id
+            domains = {}
+            domains = self._get_domains(domains, partner_id, session)
+            order_ids = sale_obj.search(domains)
+            orders = sale_obj.browse(order_ids)
+            orders.action_invoice_create(grouped=True)
+            # Dummy call to workflow, will not create another invoice
+            # but bind the new invoice to the subflow
+            orders.signal_workflow('manual_invoice')
+        return True
+
+
+class PosConfig(models.Model):
+    _inherit = 'pos.config'
+
+    anonymous_partner_id = fields.Many2one('res.partner',
```

Comment:
This field is not visible on the POS view. You must add it if we want to configure it correctly